src/compiler/checker.ts

```diff
@@ -12495,6 +12509,36 @@ namespace ts {
         }
 
         /**
+         * Checks that the return type provided is an instantiation of the global Promise<T> type

```

Comment:
@jods4 since this PR is done and closed may I suggest taking your example across to #6686 where I think it would be quite relevent.